### PR TITLE
Fix breakout pivot tether

### LINKED ISSUE #43075: Break out popover overflows the screen
### Describe the bug

The break out popover in the visualization was placed with overflow on the window (see video).

https://github.com/metabase/metabase/assets/1250185/11c7908f-eebf-433b-acab-7cf8188d89ca

### To Reproduce

1. New question -> Sample database -> People
2. Add an aggregation -> Count of rows -> Source
3. Visualize
4. Click on one of the visualization columns
5. Break out by
6. Time

### Expected behavior

The popover should be scrollable and not overflow the screen.

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:126.0) Gecko/20100101 Firefox/126.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1",
    "os.name": "Mac OS X",
    "os.version": "14.2.1",
    "user.language": "en",
    "user.timezone": "Europe/Amsterdam"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2024-05-23",
      "src_hash": "3408dbf541bb1904a57dcb2cf7022f8e76d2b1ba",
      "tag": "v0.1.6-SNAPSHOT",
      "hash": "40074bd"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

P3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/43075

Fixes an issue where the popover on the click actions in the visualization was placed with overflow on the window.

### Description

1. Adds a max height on the popover content 
2. Allow the click action popover to untether when necessary to prevent overflowing the viewport

### How to verify

1. New question -> Sample database -> People
2. Add an aggregation -> Count of rows -> Source
3. Visualize
4. Click on one of the visualization columns
5. Break out by
6. Time

The resulting popover should not overflow the screen.

### Demo
_Note: This demo artificially adds more content to the popover since we don't have big enough tables in the dev database for this to be a problem._

https://github.com/metabase/metabase/assets/1250185/2db06d8f-dde3-4edd-b6bf-dcfba31bbc21

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx [hunk 1]
```diff
@@ -169,6 +169,8 @@ export class ClickActionsPopover extends Component<
               name: "preventOverflow",
               options: {
                 padding: 16,
+                altAxis: true,
+                tether: false,
```
Comment (kamilmielnik) [inline frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx:173]: Can we add a repro test for the bug?

We could assert that `document.body.parentElement.scrollHeight` is not greater than `document.body.parentElement.offsetHeight`.

**Review by kamilmielnik — COMMENTED** [review]
Works well :+1:

**Review by ranquild — COMMENTED** [review]
Could we migrate ClickActionPopover to mantine's Popover instead?

**Comment by romeovs** [comment]
@ranquild happy to update to Mantine, but that seems like a bigger project that might also warrant updating nested components to match.

Mantine's `Popover` cannot be rendered separately from it's trigger for example, so we would need to update how the click actions work.

This was meant as a quick fix to fix the bug and didn't want to get nerd-sniped into a full rewrite/refactor. Still: happy to do that rewrite but we might need to scope that work first.

File: frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx (hunk 1 above)
Comment (romeovs) [reply frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx:173]: I've added the test, but had to use a different element to do the check on.

File: e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    cy.viewport(1000, 300);
+    cy.intercept("POST", "/api/dataset").as("dataset");
+
+    restore();
+    cy.signInAsAdmin();
+
+    createQuestion(questionDetails, { visitQuestion: true });
+  });
+
+  it("the breakthrough popover should fit within the window (metabase#43075)", () => {
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts:31]:
```suggestion
  it("the breakout popover should fit within the window (metabase#43075)", () => {
```

File: e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    cy.get("[data-testid=cell-data]").contains("54").click();
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts:32]:
```suggestion
    cy.findAllByTestId("cell-data").contains("54").click();
```

File: e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  createQuestion,
+  popover,
+  restore,
+  type StructuredQuestionDetails,
+} from "e2e/support/helpers";
+
+const { PRODUCTS, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+const questionDetails: StructuredQuestionDetails = {
+  name: "20548",
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts:12]:
```suggestion
  name: "43075",
```

Or we can skip the name altogether

```suggestion
```

File: e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+    breakout: [["field", PRODUCTS.CATEGORY, null]],
+  },
+};
+
+describe("issue 43075", () => {
+  beforeEach(() => {
+    cy.viewport(1000, 300);
+    cy.intercept("POST", "/api/dataset").as("dataset");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-charts/reproductions/43075-dropdown-overflows.cy.spec.ts:23]: Unused
```suggestion
```

**Review by kamilmielnik — APPROVED** [review]
LGTM :+1: 

Please run a stress test for the new test to make sure it does not flake: https://github.com/metabase/metabase/actions/workflows/e2e-stress-test-flake-fix.yml :pray:

**Pull request merged**